src/System.Runtime.InteropServices.RuntimeInformation/src/ArchitectureManager.Windows.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Runtime.InteropServices
+{
+    internal partial struct ArchitectureManager
+    {
+        private static Architecture ProcessArchCore
+        {
+            get
+            {
+                if (IsARM)
+                {
+                    return Architecture.ARM32;
+                }
+                else
+                {
+                    return _is32BitProcess ? Architecture.BIT32 : Architecture.BIT64;
+                }
+            }
+        }
+
+        private static Architecture OSArchCore
```

Comment:
Private functions of considerable expense (p/invoke, allocs, etc.) shouldn't be properties. It would be much clearer if the Core parts that don't handle caching were methods IMHO.